### ISSUE TITLE
Latest commit crashes with Failed to initialize EGL on NixOS using an Nvidia GPU. 

### ISSUE THREAD

**Issue by xusiai** [hyprwm/Hyprland#5317]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit 7ce781e87cf7cf789a54d37af7d78f1c11d66dbc  ().
Date: 2024-03-02
Tag: 

flags: (if any)


System Information:
System name: Linux
Node name: naiad
Release: 6.7.6-xanmod1
Version: #1-NixOS SMP PREEMPT_DYNAMIC Tue Jan  1 00:00:00 UTC 1980


GPU information: 
08:00.0 VGA compatible controller [0300]: NVIDIA Corporation GP104 [GeForce GTX 1080] [10de:1b80] (rev a1) (prog-if 00 [VGA controller])


os-release: ANSI_COLOR="1;34"
BUG_REPORT_URL="https://github.com/NixOS/nixpkgs/issues"
BUILD_ID="24.05.20240229.1536926"
DOCUMENTATION_URL="https://nixos.org/learn.html"
HOME_URL="https://nixos.org/"
ID=nixos
IMAGE_ID=""
IMAGE_VERSION=""
LOGO="nix-snowflake"
NAME=NixOS
PRETTY_NAME="NixOS 24.05 (Uakari)"
SUPPORT_URL="https://nixos.org/community.html"
VERSION="24.05 (Uakari)"
VERSION_CODENAME=uakari
VERSION_ID="24.05"


plugins:


```

</details>

### Bug or Regression?

Regression

### Description

Crashes on startup after updating from 7ce781e87cf7cf789a54d37af7d78f1c11d66dbc to fcd9d77b642c0cd45cae61cf10ed1924f2e7945b. As the crash is instant I cannot run hyprctl on the broken build. The information above is from the latest _working_ build I have.  

@fufexan 

### How to reproduce

Run Hyprland from fcd9d77b642c0cd45cae61cf10ed1924f2e7945b on NixOS with an Nvidia GPU.

### Crash reports, logs, images, videos

[hyprland.log](https://github.com/hyprwm/Hyprland/files/14798993/hyprland.log)
[hyprlandCrashReport8569.txt](https://github.com/hyprwm/Hyprland/files/14799007/hyprlandCrashReport8569.txt)

**Comment by vaxerski** [comment]
`git bisect` please

**Comment by Chemchu** [comment]
Any news on this? I just installed NixOs and tried to use Hyprland. I also have a Nvidia gpu

**Comment by Slimemaster0** [comment]
You can try [using vulkan](https://wiki.gentoo.org/wiki/Hyprland#Using_Vulkan) (Not supported)
To do this add: ``env = WLR_RENDERER,vulkan`` to you hyprland.conf

**Comment by vaxerski** [comment]
it doesn't do anything. Stop acting like it does.

**Comment by Chemchu** [comment]
C'mon man, at least he's trying to help us here. 
Anyways, I "fixed" the problem my Nvidia GPU was causing (it's more like a workaround). It seems there's some issues with the stable release of the Nvidia drivers. 
I'll leave my current solution here in case it helps.  

### This didn't work for me (not right now)
```nix
# configuration.nix
hardware.nvidia.package = config.boot.kernelPackages.nvidiaPackages.stable; # --> points to version 550
```

### This actually worked
```nix
# configuration.nix
hardware.nvidia.package = config.boot.kernelPackages.nvidiaPackages.mkDriver {
    version = "535.129.03"; # --> older but working version
    sha256_64bit = "sha256-5tylYmomCMa7KgRs/LfBrzOLnpYafdkKwJu4oSb/AC4=";
    sha256_aarch64 = "sha256-i6jZYUV6JBvN+Rt21v4vNstHPIu9sC+2ZQpiLOLoWzM=";
    openSha256 = "sha256-/Hxod/LQ4CGZN1B1GRpgE/xgoYlkPpMh+n8L7tmxwjs=";
    settingsSha256 = "sha256-QKN/gLGlT+/hAdYKlkIjZTgvubzQTt4/ki5Y+2Zj3pk=";
    persistencedSha256 = "sha256-FRMqY5uAJzq3o+YdM2Mdjj8Df6/cuUUAnh52Ne4koME=";
};
```

As I said, it's just a workaround. I guess this issue won't last for long. 
Okay bye!

**Comment by vaxerski** [comment]
> C'mon man, at least he's trying to help us here.

No, it's misleading and I have repeated 100 times that there is ***no*** vulkan renderer in hyprland, no matter what env you set.

**Comment by xusiai** [comment]
Went through git bisect and found no broken Hyprland commits. Currently using 81766647f2dc34013a62066edd191426b8f16a42 along with the closed source nvidia 550.54.14 driver. These are pinned to their versions and work. However updating the lockfile _with these two pinned to working versions_ still result in Hyprland crashing with the same log posted above. Therefore it seems to me like the culprit is another package (wlroots?) entirely. 

Regardless, this is about as far as my ability to debug goes. Hopefully someone more knowledgeable can chime in.

**Comment by vaxerski** [comment]
likely a wlroots issue so I will close this
